src/game/audio/sound_effects.rs

```diff
@@ -0,0 +1,100 @@
+
+use bevy::{ecs::world::Command, prelude::*};
+use rand::{distributions::Uniform, Rng as _};
+
+use crate::game::assets::SoundEffectHandles;
+
+pub(super) fn plugin(_app: &mut App) {
+    // No setup required for this plugin.
+    // It's still good to have a function here so that you can add some setup later if needed.
+}
+
+impl SoundEffectHandles {
+    /// Plays a random sound effect matching the given name.
+    ///
+    /// When defining the settings for this method, you almost always want to use [`PlaybackMode::Despawn`](bevy::audio::PlaybackMode).
+    /// Every time a sound effect is played, a new entity is generated. Once the sound effect is complete,
+    /// the entity should be cleaned up, rather than looping or sitting around uselessly.
+    ///
+    /// This method accepts any type which implements `AsRef<str>`.
+    /// This allows you to pass in `&str`, `String`, or a custom type that can be converted to a string.
+    ///
+    /// These custom types can be useful for defining enums that represent specific sound effects.
+    /// Generally speaking, enum values should be used to represent one-off or special-cased sound effects,
+    /// while string keys are a better fit for sound effects corresponding to objects loaded from a data file.
+    fn play(&mut self, name: impl AsRef<str>, world: &mut World, settings: PlaybackSettings) {
+        let name = name.as_ref();
+        if let Some(sfx_list) = self.get_mut(name) {
+            // If you need precise control over the randomization order of your sound effects,
+            // store the RNG as a resource and modify these functions to take it as an argument.
+            let rng = &mut rand::thread_rng();
+
+            let index = rng.sample(Uniform::from(0..sfx_list.len()));
+            // We don't need a (slightly) more expensive strong handle here (which are used to keep assets loaded in memory)
+            // because a copy is always stored in the SoundEffects resource.
+            let source = sfx_list[index].clone_weak();
+
+            world.spawn(AudioBundle {
+                source,
+                // We want the sound effect to play once and then despawn.
+                settings,
+            });
+        } else {
+            warn!("Sound effect not found: {}", name);
+        }
+    }
+}
+
+/// A custom command used to play sound effects.
+struct PlaySoundEffect {
+    name: String,
+    settings: PlaybackSettings,
+}
+
+impl Command for PlaySoundEffect {
+    fn apply(self, world: &mut World) {
+        // Access both the world and the resource we need from it using resource_scope
+        // which temporarily removes the SoundEffects resource from the world
+        world.resource_scope(|world, mut sound_effects: Mut<SoundEffectHandles>| {
+            sound_effects.play(self.name, world, self.settings);
+        });
+    }
+}
+
+/// An "extension trait" used to make it convenient to play sound effects via [`Commands`].
```

Comment:
```suggestion
/// An extension trait used to make it convenient to play sound effects via [`Commands`].
```
This really is an extension trait :smile:

Reply:
Taken from https://github.com/bevyengine/bevy/pull/14554

Reply:
Fixed in [`cabeee5` (#225)](https://github.com/TheBevyFlock/bevy_quickstart/pull/225/commits/cabeee5d8e6ea3f350cab88cda624961be500ce8), but you should add this review comment there as well.